### ISSUE TITLE
[CoE Starter Kit - BUG] AI Usage sync from environments in CoE table "admin_AICreditsUsage" is showing credits consumed only from last processed environment

### ISSUE THREAD

**Issue by Arodyx** [microsoft/coe-starter-kit#9343]
### Does this bug already exist in our backlog?

- [x] I have checked and confirm this is a new bug.

### Describe the issue

When there is a user who has a flows in multiple environments that are using AI credits at the same day, then sync is not correctly gathering credits consumed.
Flow gathers credits consumed from environment (from table msdyn_AIEvent ) and puts it in the CoE environment into admin_AICreditsUsage table with primary key:
-<userAzureADID>-<ProcessingDate>
If the same user is using credits on multiple environment, this sync will override record with the last processed environment (processing is done randomly to avoid throttling).

### Expected Behavior

admin_AICreditsUsage  table on CoE environment should have separate rows for each environment where user had consumed AI credits on each day.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.37

### What app or flow are you having the issue with?

Admin | Sync Template v4 (AI Usage)

### What method are you using to get inventory and telemetry?

Cloud flows

### Steps To Reproduce

1. In Environment1: UserA creates a flow and adds AI builder action that consumes credits (for example, detect language)
2. In Environment1: UserA runs flow few times, credits consumed are logged in msdyn_AIEvent table
3. In Environment2: UserA creates a new flow that consumes credits
4. In Environment2: UserA runs flow few times, credits consumed are logged in msdyn_AIEvent table, credits consumed is different as number of runs are different between Environment1 and Environment2
5. CoE sync is triggered (manually or automatically). it will create only 1 row in table admin_AICreditsUsage for userA with number of credits consumed only on environment that is logged

### Anything else?

It can be easily fixed by adding Environment GUID as additional parameter in 2 actions:
- In "See if in inventory": filter rows
- In "Upsert AI Credits Usage record": in Name parameter

I checked with latest CoE version (4.45.1) JSON files and above actions in the flow are the same so it seems to not been updated in never version than mine.

**Comment by Jenefer-Monroe** [comment]
You're right. Well that is awful. I did not author this flow but its pretty clear that this is an issue. 
Marking as a bug to track and fix for next release.

**Comment by pbattini** [comment]
@Arodyx Fix was part of Feb 2025 release. Closing the ticket as no further action needed. Please reach out if you have any questions.
